CoverMyDotNet/Client.cs

```diff
@@ -18,15 +19,42 @@ public Client() : base()
 			this.BaseUrl = new Uri(string.IsNullOrEmpty(apiUrl) ? 
 				"https://api.covermymeds.com" : apiUrl);
 			//try to get the api creds from the environment first
+			this.FollowRedirects = false; //we dont want to follow redirects because when we do, it does not post the correct headers
 			_apiId = Environment.GetEnvironmentVariable("CMM_API_ID");
-			_apiSecret = Environment.GetEnvironmentVariable("CMM_API_SECRET");			
+			_apiSecret = Environment.GetEnvironmentVariable("CMM_API_SECRET");		
+			//for the requestpages
+			AddHandler("application/typed+json", new RestSharp.Deserializers.JsonDeserializer());	
 		}
 
 		public Client(string apiId, string apiSecret, string apiUrl = "https://api.covermymeds.com") : base()
 		{
 			this.BaseUrl = new Uri(apiUrl);			
+			this.FollowRedirects = false;
```

Comment:
Similar to PR #3 - constructor chaining might help you DRY these assignments up.